src/waltz/mib/fd_netdev_netlink.c

```diff
@@ -168,7 +174,44 @@ fd_netdev_netlink_load_table( fd_netdev_tbl_join_t * tbl,
         }
         netdev->master_idx = (short)master_idx;
         break;
-      }
+      } /* IFLA_MASTER */
+
+      case IFLA_LINKINFO : {
+        if ( ifi->ifi_type!=ARPHRD_IPGRE ) continue;
+        
+        struct rtattr * info_rat    = rta;
+        long            info_rat_sz = (long)rta_sz;
+        
+        for( ; RTA_OK( info_rat, info_rat_sz ); info_rat = RTA_NEXT( info_rat, info_rat_sz ) ) {
+          void * info_rta    = RTA_DATA( info_rat );
+          ulong  info_rta_sz = RTA_PAYLOAD( info_rat ); 
+
+          switch( info_rat->rta_type ) {
+          case IFLA_INFO_KIND: {
+            char * kind = (char *)info_rta;
+            if( FD_UNLIKELY( strcmp( kind, "gre" ) !=0 ) ) FD_LOG_ERR(( "rta is not type gre: %s", kind ));
+            break;
+          } /* IFLA_INFO_KIND */
+          case IFLA_INFO_DATA: {
+            struct rtattr * gre_rat    = (struct rtattr *)info_rta;
+            long            gre_rat_sz = (long)info_rta_sz;
+            
+            for( ; RTA_OK( gre_rat, gre_rat_sz ); gre_rat = RTA_NEXT( gre_rat, gre_rat_sz ) ) {
+              void * gre_rta    = RTA_DATA( gre_rat );
+              uint gre_dst_ip;
+
+              switch( gre_rat->rta_type ) {
+              case IFLA_GRE_REMOTE:
+                gre_dst_ip = FD_LOAD( uint, gre_rta );
+                if( gre_dst_ip != 0 ) netdev->gre_dst_ip = gre_dst_ip;
+                break;
+              }
```

Comment:
We might want to check for IFLA_GRE_LOCAL too. 
I think the GRE local attribute is used as the source address for outgoing packets, but I'm not sure ...